File: Level 2/가장 큰 정사각형 찾기/solution.py
def solution(board):
    ROW, COL = len(board[0]), len(board)

    for i in range(1, COL):
        for j in range(1, ROW):
            if board[i][j] == 1:
                up, left, up_left = (
                    board[i - 1][j],
                    board[i][j - 1],
                    board[i - 1][j - 1],
                )

                board[i][j] = min(up, left, up_left) + 1

    max_value = max(max(row) for row in board)

    return max_value**2

File: Level 2/가장 큰 정사각형 찾기/solution.js
function solution(board) {
  const ROW = board[0].length;
  const COL = board.length;

  for (let i = 1; i < COL; i++) {
    for (let j = 1; j < ROW; j++) {
      if (board[i][j] === 1) {
        const up = board[i - 1][j];
        const left = board[i][j - 1];
        const upLeft = board[i - 1][j - 1];

        board[i][j] = Math.min(up, left, upLeft) + 1;
      }
    }
  }

  const maxValue = Math.max(...board.map((row) => Math.max(...row)));

  return maxValue ** 2;
}
